cas-server-support-oauth/src/main/java/org/jasig/cas/support/oauth/web/BaseOAuthWrapperController.java

```diff
@@ -106,24 +107,28 @@ protected OAuthWebApplicationService createService(final RegisteredService regis
      * Create an authentication from a user profile.
      *
      * @param profile the given user profile
+     * @param service the registered service
      * @return the built authentication
      */
-    protected Authentication createAuthentication(final UserProfile profile) {
```

Comment:
I understand the rational here, but eventually, this will have to be removed. What is happening in the OAuth world today is that you are validating tickets manually, which requires that you execute this kind of logic here to make sure services are authorized, and attributes are properly released, etc.

What you want to do instead is:

Ask CASImpl to validate the ST for you and give you a principal, via the callback url as the service. Once you do, you will have a CAS principal who's allowed the proper attributes automatically, and you wouldnt need duplicate logic here.